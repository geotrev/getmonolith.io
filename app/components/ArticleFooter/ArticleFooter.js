import React from "react"
import PropTypes from "prop-types"

const PRISM_LANG_JS_CLASS = "language-s"

export default function ArticleFooter({ name, hasAPI }) {
  const renderAPIExamples = (scope = false) => {
    const scopeString = scope ? "#wrapper-element" : ""

    return <pre className={PRISM_LANG_JS_CLASS}>{`Undernet.${name}.start(${scopeString})`}</pre>
  }

  const renderComponentAPI = () => {
    if (!hasAPI) return

    return (
      <>
        <h3>API</h3>

        <p>
          {
            "Call one of the following scripts from Undernet's JavaScript (not both!). This should happen only once on page load."
          }
        </p>
        <pre className={PRISM_LANG_JS_CLASS}>{"Undernet.start()"}</pre>
        {renderAPIExamples()}

        <p>
          {
            "Alternatively, if you're using a UI framework like React, you can do any of the above API calls but pass a selector string to scope Undernet and its events within the corresponding element:"
          }
        </p>
        {renderAPIExamples(true)}
      </>
    )
  }

  return (
    <>
      {renderComponentAPI()}
      <hr />
      <p className="has-text-end">
        {"Is this article inaccurate? "}
        <a
          href={`https://github.com/geotrev/undernet/tree/master/app/docs/${name.toLowerCase()}.md`}
        >
          {"Edit this page on Github!"}
        </a>
      </p>
    </>
  )
}

ArticleFooter.propTypes = {
  name: PropTypes.string.isRequired,
  hasAPI: PropTypes.bool,
}
