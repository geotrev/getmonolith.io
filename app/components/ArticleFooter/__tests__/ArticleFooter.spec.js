import React from "react"
import ArticleFooter from "../ArticleFooter"

describe("<ArticleFooter [name] />", () => {
  it("renders", () => {
    const wrapper = mount(<ArticleFooter name="JavaScript" />)
    expect(wrapper).toMatchSnapshot()
  })

  it("renders component API", () => {
    const wrapper = mount(<ArticleFooter name="Collapsible" hasAPI />)
    expect(wrapper).toMatchSnapshot()
  })
})
