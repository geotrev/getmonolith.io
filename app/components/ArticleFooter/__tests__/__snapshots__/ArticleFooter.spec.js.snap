// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleFooter [name] /> renders 1`] = `
<ArticleFooter
  name="JavaScript"
>
  <hr />
  <p
    className="has-text-end"
  >
    Is this article inaccurate? 
    <a
      href="https://github.com/geotrev/undernet/tree/master/app/docs/javascript.md"
    >
      Edit this page on Github!
    </a>
  </p>
</ArticleFooter>
`;

exports[`<ArticleFooter [name] /> renders component API 1`] = `
<ArticleFooter
  hasAPI={true}
  name="Collapsible"
>
  <h3>
    API
  </h3>
  <p>
    Call one of the following scripts from Undernet's JavaScript (not both!). This should happen only once on page load.
  </p>
  <pre
    className="language-s"
  >
    Undernet.start()
  </pre>
  <pre
    className="language-s"
  >
    Undernet.Collapsible.start()
  </pre>
  <p>
    Alternatively, if you're using a UI framework like React, you can do any of the above API calls but pass a selector string to scope Undernet and its events within the corresponding element:
  </p>
  <pre
    className="language-s"
  >
    Undernet.Collapsible.start(#wrapper-element)
  </pre>
  <hr />
  <p
    className="has-text-end"
  >
    Is this article inaccurate? 
    <a
      href="https://github.com/geotrev/undernet/tree/master/app/docs/collapsible.md"
    >
      Edit this page on Github!
    </a>
  </p>
</ArticleFooter>
`;
